engine/tests/phpunit/unit/Elgg/I18n/TranslatorUnitTest.php

```diff
@@ -182,4 +182,31 @@ public function testCanDetectCurrentLanguageFromConfig() {
 		$this->assertEquals('af', $this->translator->getCurrentLanguage());
 	}
 
+	public function testCanOverrideExistingTranslation() {
+
+		$this->assertEquals('Dummy', $this->translator->translate('__elgg_php_unit:test_key', [], 'en'));
+		$this->assertEquals('__elgg_php_unit:test_key2', $this->translator->translate('__elgg_php_unit:test_key2', [], 'en'));
+
+		$this->assertEquals('Dummy', $this->translator->translate('__elgg_php_unit:test_key', [], 'new_lang_code'));
+		$this->assertEquals('__elgg_php_unit:test_key2', $this->translator->translate('__elgg_php_unit:test_key2', [], 'new_lang_code'));
+
+		$this->translator->addTranslation('en', [
+			'__elgg_php_unit:test_key' => 'Not So Dummy',
+			'__elgg_php_unit:test_key2' => 'Still Dummy',
+		]);
+
+		$this->translator->addTranslation('new_lang_code', [
+			'__elgg_php_unit:test_key' => 'Not So Dummy',
```

Comment:
could you change the translated string to something different than that from the `en` translation? That way we are sure it is not coming from the `en` language when getting it.

Reply:
Done